sources/web/datalab/polymer/components/datalab-files/datalab-files.ts

```diff
@@ -179,13 +179,17 @@ class FilesElement extends Polymer.Element {
     clearInterval(this._fileListRefreshIntervalHandle);
   }
 
-  _getNotebookUrlPrefix() {
-    let prefix = location.protocol + '//' + location.host + '/';
+  async _getNotebookUrlPrefix() {
+    // Notebooks that are stored on the VM requires the basepath.
+    let basepath = await ApiManager.getBasePath();
```

Comment:
I think this would make sense if we were willing for location.protocol + '//' + location.host + basepath+ '/' to be there, but 'edit' and 'notebook' are specific to this file.  Currently ApiManager doesn't do anything with location.* either, so I'm going to leave it all here.